python/cudf/dataframe/series.py

```diff
@@ -689,6 +689,39 @@ def fillna(self, value, method=None, axis=None, inplace=False, limit=None):
         if not inplace:
             return self._copy_construct(data=data)
 
+    def where(self, cond, other=None, axis=None):
+        """
+        Replace values with other where the condition is False.
+
+        :param cond: boolean
+            Where cond is True, keep the original value. Where False,
+            replace with corresponding value from other.
+        :param other: scalar
+            Entries where cond is False are replaced with
+            corresponding value from other.
+        :param axis:
+        :return: Series
+
+        """
+        idx_bool = cond.to_array()
+        to_replace = self.to_array()[idx_bool]
```

Comment:
Should keep both of these on the GPU instead of converting to numpy arrays in host memory.

Reply:
That was a kind of draft. Please ignore them.